### ISSUE TITLE
Support for a pre-'docker build' command

### ISSUE THREAD

**Issue by jamiegs** [Microsoft/DockerTools#68]
I'd like the ability to run a command prior to Tools for Docker building the containers.  The command is used to setup our developer's environment.  Maybe such a feature exists but I couldn't find it in any documentation.

It needs to happen before the container build because I'm writing out a .env file to pass some user-specific values in a .env file that get read into docker-compose.yml.

It would be nice to have the ability to run a command after debug stop also.

**Comment by dbreshears** [comment]
So for running a command prior to building the containers,  you should be able to go to Tools->Options->Container Tools->Settings and uncheck "Automatically start containers in the background".     Then go to the web or console project you have added Docker support to, right click and go to properties,   Open Build Events tab and add your Pre-build event command line.

We don't have an extensibility point I can think of currently to execute a command when you stop debugging.

**Comment by jamiegs** [comment]
Thanks, I was able to get it working, however it seems that the "Automatically start containers in the background" setting needs set per user.. which will require all of our users to uncheck the box or they'd have some unexpected results. 

Is there any way similar functionality, like PreBuildEvents, added to the dcproj file as we'd only want the command to run if the user chose to debug using docker.. if they chose to run outside of docker for some reaon we don't need the command to run.

**Comment by dbreshears** [comment]
Unfortunately, we don't have support in the .dcproj currently.    Your option at this point would be to update our Docker MSBuild .targets file to execute your command at appropriate time,  but then beware that we would likely overwrite your custom targets each time we release new tooling.

**Comment by bwateratmsft** [comment]
This issue is being closed as an inactive issue. If you wish to keep it active, please re-open with any additional context for our consideration, we are happy to take a look!
